diff --git a/tcmalloc/BUILD b/tcmalloc/BUILD
--- a/tcmalloc/BUILD
+++ b/tcmalloc/BUILD
@@ -24,7 +24,7 @@
 load("//tcmalloc:copts.bzl", "TCMALLOC_DEFAULT_COPTS", "TCMALLOC_DEFAULT_CXXOPTS")
 load("//tcmalloc:variants.bzl", "create_tcmalloc_benchmark", "create_tcmalloc_libraries", "create_tcmalloc_testsuite")
 
-package(default_visibility = ["//visibility:private"])
+package(default_visibility = ["//visibility:public"])
 
 licenses(["notice"])
 
